hddfancontrol/__init__.py

```diff
@@ -974,19 +974,22 @@ def set_high_priority(logger: logging.Logger) -> None:
     # use "real time" scheduler
     done = False
     sched = os.SCHED_RR
-    if os.sched_getscheduler(0) == sched:
```

Comment:
Catching OSError may be too broad, I would prefer something like (untested code):
```python
try:
    eff_sched = os.sched_getscheduler(0)
except OSError as e:
    if e.errno != errno.ENOSYS:
        raise
    logger.warning("sched_getscheduler is not supported on this system, leaving scheduler as is")
else:
    ...
```

Reply:
Obviously you need to import `errno`. Please test on your system to confirm that works.